feat(templates, cli, docs-server): update to handle beta

DanielSLew opened the pull request:
## Description

- Templates now don't dynamically create a branch for every provider, each template should be agnostic and just use a env var to set the models until the user wants to set it otherwise.
- MCP docs server will install the beta version of the docs server if they create a project with the `beta` tag.
- Updates to the templates now will get pushed to the beta branch, when beta goes stable we will merge the beta branch into the main branch for all templates and update the github script to push to `main`.
- Templates have been cleaned up
- small docs updates based off of how the template migrations went

Follow ups:
- PR in the 0.x branch to add a migrationTool to the MCP docs server with the same prompt names that when run will tell the user to update their mcp docs server to use the `@beta` tag in order to get the up to date migration information.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Beta release channel enabled; CLI and templates surface branch selection and explicit provider/model selection via a new MODEL environment variable.

* **Bug Fixes**
  * More consistent error handling and safe fallbacks across tools and workflows (PDF, extraction, image/audio, Q&A, CSV).

* **Documentation**
  * READMEs, migration guide, and .env examples updated with multi-provider model instructions and beta upgrade guidance.

* **Tests**
  * Expanded tests for cloning, branch handling, provider selection, and template creation.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

DanielSLew committed:
feat(templates, cli, docs-server): update to handle beta

DanielSLew committed:
update templates to beta

DanielSLew committed:
update template builder code

DanielSLew committed:
clean up

DanielSLew committed:
merge main

LekoArts commented on docs/src/content/en/guides/migrations/upgrade-to-v1/overview.mdx:
```diff
@@ -13,7 +13,7 @@ Need help with the migration? Join our [Discord community](https://discord.gg/BT
-### Update all Mastra packages
+### Update all Mastra packages to `beta` tag
 
 Use your package manager to update your project's versions. Be sure to update **all** Mastra packages at the same time to ensure compatibility.
```
Good idea to clarify this 👍 

````suggestion
### Update all Mastra packages to `beta` tag

Use your package manager to update your project's versions. Be sure to update **all** Mastra packages at the same time to ensure compatibility.

Here's how you update the most common used packages:

```shell
npm install @mastra/core@beta @mastra/loggers@beta @mastra/memory@beta
```

Install any other Mastra package with the `@beta` tag and you'll get the latest beta version available.
````

DanielSLew committed:
fix(cli): add quotes to model identifiers in generated code

Fixes issue where model identifiers were not quoted in generated
template code, resulting in invalid TypeScript syntax.

Related: https://github.com/mastra-ai/mastra/pull/10036#discussion_r1898765432

DanielSLew committed:
docs(templates): update .env.example comments to be provider-agnostic

Changed 'Add your real OpenAI API key' to 'Add your API key for your
chosen provider' to reflect multi-provider support.

Related: https://github.com/mastra-ai/mastra/pull/10036#discussion_r1898765433

DanielSLew committed:
docs(template-pdf-questions): update feature list to reflect multi-provider support

Updated 'Single API Setup' to 'Multi-Provider Support' and removed
hardcoded OpenAI reference to align with provider-agnostic approach.

Related: https://github.com/mastra-ai/mastra/pull/10036#discussion_r1898765434

DanielSLew committed:
feat(scripts): add validation for required environment variables

Add validation to check that all required environment variables
(ORGANIZATION, GITHUB_TOKEN, USERNAME, EMAIL) are set before
running the sync script, providing clearer error messages.

Related: https://github.com/mastra-ai/mastra/pull/10036#discussion_r1898765435

DanielSLew committed:
refactor(template-deep-research): add explicit null checks for agent retrieval

Add defensive null checking for agent retrieval in deep-research
tools to provide clearer error messages instead of relying on
non-null assertions that could cause runtime TypeErrors.

Related: https://github.com/mastra-ai/mastra/pull/10036#discussion_r1898765436

DanielSLew committed:
fix test

DanielSLew committed:
changeset

DanielSLew committed:
refactor(cli): export and reuse LLM_PROVIDERS constant

Export LLM_PROVIDERS from utils.ts and reuse it in create.ts
to avoid duplication and ensure consistency.

Related: https://github.com/mastra-ai/mastra/pull/10036#discussion_r1898851234

DanielSLew committed:
test(cli): add LLM_PROVIDERS to mock in create.test.ts

Fix test failures caused by the newly exported LLM_PROVIDERS constant
not being included in the mock for ../init/utils.

Fixes CI test failures.

DanielSLew committed:
fix e2e test

DanielSLew committed:
Merge branch 'main' into feat/templates/cli/docs-server/update-to-handle-beta
